### PR TITLE
Make Scenic configurable

### PR DESCRIPTION
**Pull request by chenshay**

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -57,7 +58,11 @@ if (console.log) {
 /**
  * List all Articles.
  */
-app.get('/', (req, res) => {
+app.get(['/','/countrycode/:countrycode'], (req, res) => {
```
Comment (dvoytenko) [inline app/app.js]: Do we need "/countrycode/" text? Also, I'd rather make it more generic than just "country". Any kind of experiment/country/special-case setup could be delivered this way via Scenic.

Comment (chenshay) [reply app/app.js]: done

File: app/app.js [hunk 2]
```diff
@@ -422,3 +422,21 @@ function ampJsUrl(name, rtv) {
...
+  console.log('Testing Scenic for this country: ' + config.name);
+  console.log('Testing Scenic with this Publication: ' + getPublicationId());
+}
+
+/**
+ * @return {string}
+ */
+
+function getPublicationId() {
+  return process.env.SERVE_PUBID || 'com.appspot.scenic-2017-test';
```
Comment (dvoytenko) [inline app/app.js]: Let's just use the actual publication id - `scenic-2017.appspot.com`

File: app/app.js [hunk 3]
```diff
@@ -50,14 +50,15 @@ const ARTICLES = require('./content').ARTICLES;
...
-  console.log('Scenic started. Publication: ' + PUBLICATION_ID);
+  console.log('Scenic started. Publication: ' + getPublicationId());
 }
 
 
 /**
  * List all Articles.
  */
-app.get('/', (req, res) => {
+app.get(['/','/config/:configId'], (req, res) => {
```
Comment (dvoytenko) [inline app/app.js]: LG, but the same question: why do we need "/config/" in the path? Could we just do `/:configId/...` URLs?

File: app/app.js [hunk 4]
```diff
@@ -97,7 +96,9 @@ app.get('/landing-gpay.html', (req, res) => {
  * An Article.
  * TODO(dvoytenko): remove "/examples/" path
  */
-app.get(['/((\\d+))', '/examples/sample-pub/((\\d+))'], (req, res) => {
+app.get(['/config/:configId/((\\d+))', '/((\\d+))',
+    '/examples/sample-pub/((\\d+))'], (req, res) => {
```
Comment (dvoytenko) [inline app/app.js]: I think it'd be fine to now remove `/examples/sample-pub` URLs - they were here only for transition.

File: app/app.js [hunk 5]
```diff
@@ -422,3 +422,21 @@ function ampJsUrl(name, rtv) {
       'http://localhost:8001/dist/v0/' + name + '-0.1.max.js' :
       cdnBase + '/v0/' + name + '-0.1.js';
 }
+
+/**
+ * @param {string} id 
+ */
+function setConfig(id) {
+  const config = getConfig(id);
+  process.env.SERVE_PUBID = config.publicationId;
```
Comment (dvoytenko) [inline app/app.js]: We can't be overriding the static shared state with a request state - somewhere it'll really hurt us.

File: app/app.js [hunk 6]
```diff
@@ -106,7 +107,7 @@ app.get(['/((\\d+))', '/examples/sample-pub/((\\d+))'], (req, res) => {
   res.render('../app/views/article', {
     swgJsUrl: SWG_JS_URLS[setup.script],
     setup: setup,
-    publicationId: PUBLICATION_ID,
+    publicationId: getPublicationId(),
```
Comment (dvoytenko) [inline app/app.js]: Why not just pass the whole config object here?

File: app/app.js (hunk 5 above)
Comment (chenshay) [reply app/app.js]: what do you suggest we do? This needs to be updated in multiple files right?

File: app/app.js (hunk 3 above)
Comment (chenshay) [reply app/app.js]: done

File: app/app.js (hunk 4 above)
Comment (chenshay) [reply app/app.js]: done

File: app/app.js (hunk 2 above)
Comment (chenshay) [reply app/app.js]: done

File: app/app.js (hunk 6 above)
Comment (chenshay) [reply app/app.js]: done

File: app/app.js (hunk 5 above)
Comment (chenshay) [reply app/app.js]: fixed

File: app/config.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Properties:
+ * - sku: Required. Sku to add to the user's subscriptions.
```
Comment (dvoytenko) [inline app/config.js]: Remove

File: app/app.js [hunk 8]
```diff
@@ -421,4 +417,4 @@ function ampJsUrl(name, rtv) {
   return AMP_LOCAL ?
       'http://localhost:8001/dist/v0/' + name + '-0.1.max.js' :
       cdnBase + '/v0/' + name + '-0.1.js';
-}
+}
```
Comment (dvoytenko) [inline app/app.js]: nit: new line

File: app/config.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+ *  @typedef {{
+ *    name: string,
+ *    countryCode: string,
+ *    publicationId: string,
+ * }}
+ */
+let Config;
+
+/**
+ * @param {string} id
```
Comment (dvoytenko) [inline app/config.js]: Mark as optional since you call it w/o `id` in many cases.

File: app/views/article.html [hunk 10]
```diff
@@ -73,7 +73,7 @@
       "isPartOf": {
         "@type": ["CreativeWork", "Product"],
         "name" : "Scenic News",
-        "productID": "<% publicationId %>:news"
+        "productID": "<% config.publicationId %>:news"
```
Comment (dvoytenko) [inline app/views/article.html]: Let's also add `config.name` in the title.

File: app/config.js (hunk 7 above)
Comment (chenshay) [reply app/config.js]: done

File: app/config.js (hunk 9 above)
Comment (chenshay) [reply app/config.js]: done

File: app/views/article.html (hunk 10 above)
Comment (chenshay) [reply app/views/article.html]: done

**Pull request merged**